### ISSUE TITLE
Deploying when cloudformation stack is in bad state causes error `Cannot read property 'OutputValue' of undefined`

### ISSUE THREAD

**Issue by activescott** [serverless/serverless#1638]
Related to #1615 and #1614 

Bug Report Template -----------------------------------------------

##### Serverless Framework Version: 
1.0.0-alpha.1

##### Operating System: 
osx 10.11.5 (15F34)

##### Expected Behavior:
a more actionable error message indicating the cloudformation stack is in a bad state

##### Actual Behavior:
> Cannot read property 'OutputValue' of undefined

##### Stack-Trace (Optional):

    TypeError: Cannot read property 'OutputValue' of undefined
        at /usr/local/lib/node_modules/serverless/lib/plugins/aws/index.js:100:55
        at tryCatcher (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/util.js:16:23)
        at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:504:31)
        at Promise._settlePromise (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:561:18)
        at Promise._settlePromise0 (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:606:10)
        at Promise._settlePromises (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:685:18)
        at Async._drainQueue (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:138:16)
        at Async._drainQueues (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:148:10)
        at Immediate.Async.drainQueues [as _onImmediate] (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:17:14)
        at processImmediate [as _immediateCallback] (timers.js:383:17)

Feature Request Template ------------------------------------------

##### Benefits:

* Benefit one...

##### Drawbacks:

* Drawback one...

##### Additional Details:

Discussion template  ----------------------------------------------

##### TLDR:
This would least gets an actionable error message like follows:

> Stack:arn:aws:cloudformation:...........
     is in UPDATE_ROLLBACK_FAILED state and can not be updated.

**Comment by pmuens** [comment]
Related to https://github.com/serverless/serverless/issues/1574

Will be fixed with https://github.com/serverless/serverless/pull/1639
